forc-plugins/forc-node/src/chain_config.rs

```diff
@@ -0,0 +1,574 @@
+    /// contents of the remote configuration repo at:
+    /// https://github.com/FuelLabs/chain-configuration/
+    async fn check_github_files(
+        &self,
+        conf: &ChainConfig,
+    ) -> anyhow::Result<Vec<GithubContentDetails>> {
+        let folder_name = match conf {
+            ChainConfig::Local => LOCAL_CONFIG_FOLDER_NAME,
+            ChainConfig::Testnet => TESTNET_CONFIG_FOLDER_NAME,
+            ChainConfig::Ignition => IGNITION_CONFIG_FOLDER_NAME,
+        };
+        let api_endpoint = self.build_api_endpoint(folder_name);
+
+        let response = self
+            .client
+            .get(&api_endpoint)
+            .header("User-Agent", "forc-node")
+            .send()
+            .await?;
+
+        if !response.status().is_success() {
+            bail!("failed to fetch updates from github")
+        }
+
+        let contents: Vec<GithubContentDetails> = response.json().await?;
+        Ok(contents)
+    }
+
+    /// Calculates the hash for the local configuration instance.
+    /// The hash calculation is based on github's hash calculation to match the
+    /// github api response.
+    fn check_local_files(&self, conf: &ChainConfig) -> Result<Option<HashMap<String, String>>> {
+        let folder_name = match conf {
+            ChainConfig::Local => bail!("Local configuration should not be checked"),
+            ChainConfig::Testnet => TESTNET_CONFIG_FOLDER_NAME,
+            ChainConfig::Ignition => IGNITION_CONFIG_FOLDER_NAME,
+        };
+
+        let folder_path = self.config_vault.join(folder_name);
+
+        if !folder_path.exists() {
+            return Ok(None);
+        }
+
+        let mut files = HashMap::new();
+        for entry in std::fs::read_dir(&folder_path)? {
+            let entry = entry?;
+            if entry.path().is_file() {
+                let content = std::fs::read(entry.path())?;
+                // Calculate SHA1 the same way github does
+                let mut hasher = Sha1::new();
+                hasher.update(b"blob ");
+                hasher.update(content.len().to_string().as_bytes());
+                hasher.update([0]);
+                hasher.update(&content);
+                let sha = format!("{:x}", hasher.finalize());
+
+                let name = entry.file_name().into_string().unwrap();
+                files.insert(name, sha);
+            }
+        }
+
+        Ok(Some(files))
+    }
+
+    /// Checks if a fetch is required by comparing the hashes of individual files
+    /// of the given chain config in the local instance to the one in github by
+    /// utilizing the github content abi.
+    pub async fn check_fetch_required(&self, conf: &ChainConfig) -> anyhow::Result<bool> {
+        if *conf == ChainConfig::Local {
+            return Ok(false);
+        }
+
+        let local_files = match self.check_local_files(conf)? {
+            Some(files) => files,
+            None => return Ok(true), // No local files, need to fetch
+        };
```

Comment:
nit: could warn?
```suggestion
        // If no local files exist (None), we need to fetch. Otherwise use existing files.
        let local_files = self.check_local_files(conf)?.unwrap_or_else(|| {
            println_warning("No local configuration files found for {}", conf);
            return Ok(true);
        });
```

Reply:
I already do a warnings in the upper level function that calls this so adding here will be duplicate